java/src/main/java/com/logicalclocks/hsfs/metadata/StatisticsApi.java

```diff
@@ -0,0 +1,153 @@
+
+    String statisticsJson = hopsworksClient.getObjectMapper().writeValueAsString(statistics);
+    HttpPost postRequest = new HttpPost(uri);
+    postRequest.setHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+    postRequest.setEntity(new StringEntity(statisticsJson));
+
+    LOGGER.info("Sending metadata request: " + uri);
+    LOGGER.info(statisticsJson);
+
+    return hopsworksClient.handleRequest(postRequest, Statistics.class);
+  }
+
+  public Statistics get(FeatureGroup featureGroup, String commitTime) throws FeatureStoreException, IOException {
+    return get(featureGroup.getFeatureStore().getProjectId(), featureGroup.getFeatureStore().getId(),
+      featureGroup.getId(), commitTime);
+  }
+
+  public Statistics get(TrainingDataset trainingDataset, String commitTime) throws FeatureStoreException, IOException {
+    return get(trainingDataset.getFeatureStore().getProjectId(), trainingDataset.getFeatureStore().getId(),
+      trainingDataset.getId(), commitTime);
+  }
+
+  private Statistics get(Integer projectId, Integer featurestoreId, Integer entityId, String commitTime)
+      throws FeatureStoreException, IOException {
+    HopsworksClient hopsworksClient = getInstance();
+    String pathTemplate = PROJECT_PATH
+        + FeatureStoreApi.FEATURE_STORE_PATH
+        + STATISTICS_PATH + "?"
+        + FILTER_COMMIT_TIME_EQ + "&"
+        + CONTENT_FIELD;
+
+    String uri = UriTemplate.fromTemplate(pathTemplate)
+        .set("projectId", projectId)
+        .set("fsId", featurestoreId)
+        .set("entityType", entityType.getValue())
+        .set("entityId", entityId)
+        .set("commitTime", commitTime)
+        .expand();
+
+    LOGGER.info("Sending metadata request: " + uri);
+    HttpGet getRequest = new HttpGet(uri);
+    Statistics statistics = hopsworksClient.handleRequest(getRequest, Statistics.class);
+
+    // currently getting multiple commits at the same time is not allowed
+    if (statistics.getItems().size() == 1) {
+      return statistics.getItems().get(0);
+    }
+    return null;
+  }
+
+  public Statistics getLast(FeatureGroup featureGroup) throws FeatureStoreException, IOException {
+    return getLast(featureGroup.getFeatureStore().getProjectId(), featureGroup.getFeatureStore().getId(),
+      featureGroup.getId());
+  }
+
+  public Statistics getLast(TrainingDataset trainingDataset) throws FeatureStoreException, IOException {
+    return getLast(trainingDataset.getFeatureStore().getProjectId(), trainingDataset.getFeatureStore().getId(),
+      trainingDataset.getId());
+  }
+
+  private Statistics getLast(Integer projectId, Integer featurestoreId, Integer entityId)
+      throws FeatureStoreException, IOException {
+    HopsworksClient hopsworksClient = getInstance();
+    String pathTemplate = PROJECT_PATH
+        + FeatureStoreApi.FEATURE_STORE_PATH
+        + STATISTICS_PATH + "?"
+        + SORT_BY_COMMIT_TIME_DESC + "&"
+        + OFFSET + "&"
+        + LIMIT + "&"
+        + CONTENT_FIELD;
```

Comment:
Same as above.

Reply:
fixed, pls check if it is the way you meant